bug(task): keeper fails claimed->start transition — agent name mismatch on task-109

## 증상

실시간 로그에서 task FSM 전이 실패:

```
[ERROR] [Task] task transition failed: Invalid transition: claimed -> start (task-109)
[WARN] [Keeper] tool masc_transition returned error result (1/3)
```

## 분석

- FSM 위치: `lib/room/room_task.ml:396`
- `Claimed -> Start` 전이는 `assignee = agent_name` 가드가 있음
- keeper가 다른 agent가 claim한 task를 start 시도한 것

## 가능한 원인

1. keeper가 자신이 claim하지 않은 task에 `start` 전이를 호출
2. claim 후 agent_name이 변경됨 (세션 재시작 등으로 이름 불일치)
3. keeper의 masc_transition 호출에서 agent_name 파라미터 오류

## 제안

- 에러 메시지에 `expected_assignee`와 `actual_agent_name`을 포함하여 디버깅 용이하게
- keeper 턴 로직에서 claim 상태 확인 후 start 호출하도록 가드 추가

--- a/lib/room/room_task.ml
+++ b/lib/room/room_task.ml
@@ -421,10 +421,15 @@ let transition_task_r config ~agent_name ~task_id ~action
                        | Types.Release, Types.Claimed { assignee; _ }
                        | Types.Release, Types.InProgress { assignee; _ } when assignee = agent_name || force ->
                            Ok (Types.Todo, None)
+                       | Types.Start, Types.Claimed { assignee; _ } ->
+                           Error (Types.TaskInvalidState
+                             (Printf.sprintf
+                               "Cannot start %s: claimed by '%s', caller is '%s'"
+                               task_id assignee agent_name))
                        | _ ->
                            Error (Types.TaskInvalidState
-                             (Printf.sprintf "Invalid transition: %s -> %s (%s)"
-                               (task_status_to_string task.task_status) action_s task_id))
+                             (Printf.sprintf "Invalid transition: %s -> %s (%s, agent=%s)"
+                               (task_status_to_string task.task_status) action_s task_id agent_name))
                      in
                      (match transition with
                       | Error e -> Error e

